routing/car_model.hpp

```diff
@@ -1,16 +1,29 @@
 #pragma once
-
 #include "vehicle_model.hpp"
 
+#include "std/shared_ptr.hpp"
+
 namespace routing
 {
 
 class CarModel : public VehicleModel
 {
+public:
```

Comment:
If you have CarModelFactory, you should use it for CarModel instantiation, isn't it? Also, you created a factory method named AllLimitsInstance. The question is the same - why constructor is in the public section? Move it to private, and declare factory as a friend of this class.